include/llama/Array.hpp

```diff
@@ -89,6 +89,56 @@ namespace llama
 
     template <typename First, typename... Args>
     Array(First, Args... args) -> Array<First, sizeof...(Args) + 1>;
+
+    template <typename T, std::size_t N>
+    LLAMA_FN_HOST_ACC_INLINE constexpr auto push_front(Array<T, N> a, T v) -> Array<T, N + 1>
```

Comment:
Is passing by value intended here?

Reply:
Yes. While generally passing large arrays per value might cause expensive copies, the arrays in LLAMA are quite small usually. Most oftenly, these arrays hold the index space dimensions so usually just have 1 or maybe 2 or 3 elements. Furthermore, the call is forced inline as well, so a function call should never occur.

Reply:
Okay. Another question: What is the intended use case? In the test you just pass in some rvalues. Is this always the case in the real world? If not, is it intended to have two arrays after the operation?

Reply:
The way it's currently used is to add a `size_t` to an array of `size_t`s. The old and the new array continue living.